testsuites/syncgateway/functional/tests/test_log_rotation_new.py

```diff
@@ -0,0 +1,609 @@
+            "for ((i=1;i <= 2000;i += 1)); do curl -s -H 'Accept: text/plain' http://localhost:4985/db/ > /dev/null; done")
+
+        # Verify num of log files for every log file type
+        for log in SG_LOGS:
+            file_name = "/tmp/sg_logs/{}.log".format(log)
+            _, stdout, _ = remote_executor.execute("ls /tmp/sg_logs/ | grep {} | wc -l".format(log))
+            log_info("Checking for {} files for {}".format(i + 1, file_name))
+            assert stdout[0].rstrip() == str(i + 1)
+
+            sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+
+            # Generate an empty log file with size ~100MB
+            file_size = 100 * 1024 * 1024
+            sg_helper.create_empty_file(cluster_config=cluster_conf, url=sg_one_url, file_name=file_name, file_size=file_size)
+
+    sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=sg_conf)
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+
+
+@pytest.mark.syncgateway
+@pytest.mark.logging
+@pytest.mark.parametrize("sg_conf_name", ["log_rotation_new"])
+def test_invalid_logKeys_string(params_from_base_test_setup, sg_conf_name):
+    """Negative test to verify that we are not able start SG when
+    logKeys is string
+    """
+    cluster_conf = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    log_info("Using cluster_conf: {}".format(cluster_conf))
+    log_info("Using sg_conf: {}".format(sg_conf))
+
+    cluster_helper = ClusterKeywords()
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_conf)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_ip = host_for_url(sg_admin_url)
+
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("Test NA for SG  < 2.1")
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # read sample sg_conf
+    sg_one_url = cluster_hosts["sync_gateways"][0]["public"]
+
+    data = load_sync_gateway_config(sg_conf, cluster_hosts["couchbase_servers"][0], cluster_conf)
+
+    # set logKeys as string in config file
+    data['logging']["console"]["log_keys"] = "ABCD"
+    # create temp config file in the same folder as sg_conf
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp, indent=4)
+
+    # Stop sync_gateways
+    log_info(">>> Stopping sync_gateway")
+    sg_helper = SyncGateway()
+    sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+    try:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+    except ProvisioningError:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=sg_conf)
+        # Remove generated conf file
+        os.remove(temp_conf)
```

Comment:
no need to remove temp_conf. If any test uses this file, it overrides it

Reply:
Sure, added it as a clean up

Reply:
I do not see this change. Is it done?

Reply:
I left it there for better clean up, its harmless.